File: 1823-determine-if-string-halves-are-alike/determine-if-string-halves-are-alike.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        def count_vowels(string):
            vowels = set('aeiouAEIOU')
            return sum(1 for char in string if char in vowels)

        length = len(s)
        mid_point = length // 2

        first_half = s[:mid_point]
        second_half = s[mid_point:]

        return count_vowels(first_half) == count_vowels(second_half)

File: 1823-determine-if-string-halves-are-alike/determine-if-string-halves-are-alike.js
var halvesAreAlike = function(s) {
    const countVowels = (str) => {
        const vowels = new Set(['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']);
        return Array.from(str).filter(char => vowels.has(char)).length;
    };

    const length = s.length;
    const midPoint = length / 2;

    const firstHalf = s.substring(0, midPoint);
    const secondHalf = s.substring(midPoint);

    return countVowels(firstHalf) === countVowels(secondHalf);
};